lib/ansible/template/__init__.py

```diff
@@ -709,6 +765,7 @@ def set_temporary_context(self, **kwargs):
         Use a keyword that maps to the attr you are setting. Applies to ``self.environment`` by default, to
         set context on another object, it must be in ``mapping``.
         """
+        # NOTE remove in favor of copy_with_new_env?
```

Comment:
`copy_with_new_env` feels a little heavy handed for the use case we had with the `set_temporary_context`, but I'm not tied to it.

Reply:
Yeah, that was just a thought I captured in the code to re-evaluate.